contracts/core/Dispatcher.sol

```diff
@@ -414,20 +387,26 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
      * @param packet The IbcPacket data for the timed-out packet
      * @param proof The non-membership proof data needed to verify the packet timeout
      */
-    function timeout(IbcPacketReceiver receiver, IbcPacket calldata packet, Ics23Proof calldata proof) external {
+    function timeout(IbcPacketReceiver receiver, IbcPacket calldata packet, Ics23Proof calldata proof, uint256 clientId)
+        external
+    {
         // verify `receiver` is the original packet sender
         if (!portIdAddressMatch(address(receiver), packet.src.portId)) {
-            revert receiverNotIndtendedPacketDestination();
+            revert IBCErrors.receiverNotIndtendedPacketDestination();
         }
 
         // prove absence of packet receipt on Polymer chain
         // TODO: add non membership support
-        consensusStateManager.verifyNonMembership(proof, 'packet/receipt/path');
+        LightClient lightClient = clientIdToLightClient[clientId];
+        if (address(lightClient) == address(0) || clientId == 0) {
+            revert IBCErrors.clientNotFound();
+        }
```

Comment:
this code seems to be used in different places. Maybe we wanna move it to its own function or library?

Reply:
Yes! in this branch we were running up against the contract size limit - so I didn't abstract these into different functions (IIRC adding functions increases the contract size); they can't be a library because these require access to state and Libraries can't hold stated.

But if we can change this code use a library instead of inherit from it and get the typechain integration working, we can definitely clean this up